### PR TITLE
Adds a very rare sussy firelock

### PR DESCRIPTION
**Pull request by Incoming5643**
## About The Pull Request

On a VERY low chance (1/25000) a normal full tile firelock can instead be generated with a more suspicious appearance based on [this recent quality shitpost](https://www.reddit.com/r/SS13/comments/tkw5pz/go_on_crowbar_the_firelock_see_what_happens/).

https://user-images.githubusercontent.com/4176358/160050612-9d99c0ae-d580-46fb-8498-2dc95ac8a3da.mov

## Why It's Good For The Game

Sussy firelocks are completely identical when open (save the desc) and thus are apt not to be noticed except during exciting times. It's a fully cosmetic joke that got me to log into github, and that's an accomplishment in itself. Probability rate fully negotiable, I've added a one in a million chance before after all.

### CODE REVIEW

**Comment by ArcaneDefence** [comment]
Thanks, I hate it.

**Comment by Serxule** [comment]
EUQOxs_XsAArWV8

**Comment by Fikou** [comment]
image

**Comment by Watermelon914** [comment]
funny

**Comment by LemonInTheDark** [comment]
Fucking rip incoming.
~1000 seems a bit low given how many firelocks generate each round, guess you'd need to trip them to see~

**Comment by Incoming5643** [comment]
I legitimately forgot April Fools was in a couple of days but I guess that explains my sudden instinctual desire to add painful jokes to the game again.

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -74,6 +74,10 @@
 
 	check_atmos()
 
+	if(prob(0.1) && icon == 'icons/obj/doors/Doorfireglass.dmi')
```
Comment (LemonInTheDark) [inline code/game/machinery/doors/firedoor.dm:77]: why is the capitalization here fucked?

File: code/game/machinery/doors/firedoor.dm [hunk 2]
```diff
@@ -74,6 +74,10 @@
...
+		base_icon_state = "sus"
+		desc += " This one looks a bit sus..."
```
Comment (LemonInTheDark) [inline code/game/machinery/doors/firedoor.dm:79]: speaks for itself almost honestly, think you need this?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
I know this is a meme but I do kinda wanna merge it, the idea of like 1 in a hundred rounds having a cult of people finding a weird airlock is peak soul to me. (Maybe that's just what admins are for tho)

Would need to see the chance lowered a bit, still not totally sure

File: code/game/machinery/doors/firedoor.dm (hunk 1 above)
Comment (Incoming5643) [reply code/game/machinery/doors/firedoor.dm:77]: no idea, works all the same, if there were other changes I could fix it

File: code/game/machinery/doors/firedoor.dm (hunk 2 above)
Comment (Incoming5643) [reply code/game/machinery/doors/firedoor.dm:79]: It's mostly there if by some miracle people examine open firelocks. The discovery of finding something cursed before it itself can reveal itself is pretty fun too.

**Comment by Incoming5643** [comment]
let me do some firelock counts on stations and see what an effective 1 in a 100 chance would actually translate to.

**Comment by Incoming5643** [comment]
Quick and dirty per map counts (real number slightly higher because I'm just reading raw map files)

At prob(0.1)
Map | Rough Number of full tile firedoors | Effective chance of at least one traitor amogus somewhere on station

Delta Station |  410 |  ~33.7%
Ice Box Station | 293 |  ~25.4%
Kilo Station |  311 |  ~26.7%
Meta Station | 346 |  ~29.2%
Tram Station | 283 | ~24.6%

Average ~330 fire doors on a station, for a true one in a hundred experience we're looking at a prob(0.0032) check.

Keep in mind though that this assume that every firelock is engaged in every round, which definitely isn't true. I'm going to charitably round up to prob(0.004) (aka 1 in 25000) to mitigate this

At prob(0.004)

Delta Station |  410 |  ~1.6%
Ice Box Station | 293 |  ~1.2%
Kilo Station |  311 |  ~1.2%
Meta Station | 346 |  ~1.4%
Tram Station | 283 | ~1.1%

Keep in mind these are again rough numbers, and don't allow for centcom/mining doors (in so far that those matter) or someone continually rebuilding fire doors praying to RNG

**Review by Twaticus — CHANGES_REQUESTED** [review]
sus_opening animation opens like the classic firelock where sus_closing animation has the door rise from the floor. they should have a consistent animation.

either one will work but i personally like the floor rising one

**Comment by Incoming5643** [comment]
Glad someone noticed. Not an accident or anything,  the closing animation is them rising from the vent, and the opening animation is them getting killed. It's part of the Easter egg but if people really care it can be changed

**Comment by Twaticus** [comment]
oh sorry i never played amongus
if others think the reference works then its fine

**Comment by BlueMemesauce** [comment]
I think it would be better if it just had an low chance of appearing once the airlock is flipped down, but going away  when its flipped back up. That way there can't really be any cults of amogus because anyone could just flip it up again and its gone.

**Comment by LemonInTheDark** [comment]
Having a cult is half the point

**Comment by BlueMemesauce** [comment]
But amogus is something that the admins are working to prevent, and this would just encourage that. If it's only an easter egg for a couple seconds, then it wouldnt cause as much problems .You would only be able to point and say sus once before it quickly disappears. That would be much better than an amogus cult gathered around it constantly saying sus.

**Comment by bluetshirtguy** [comment]
inb4 ban baited

**Comment by Incoming5643** [comment]
I've never played Among Us either, I just think it's a funny meme :dab:

**Comment by cacogen** [comment]
You're never going to see this. Shows up in 1% of rounds, with no guarantee the firelock in question will be both tripped and seen?

**Comment by Incoming5643** [comment]
You have a 1 in a million chance of winning a pulse rifle from the arcade machine and that was one of mine too. Discovering the rare through oppressive odds is a design philosophy I'm a big fan of. So if you wanna see something sus go looking. Or you know, start a lot of fires.

**Comment by cacogen** [comment]
>You have a 1 in a million chance of winning a pulse rifle from the arcade machine and that was one of mine too.

Yeah, that's rare to the point it may as well not even exist too.

The goal should be to find the correct probability, where it shows up enough that people actually get to see it but not enough that it loses its novelty.

Maybe appearing in 1/4 of rounds was too much, but 1% is too little given there's no guarantee it'll even be seen.

**Pull request merged**